### PR TITLE
Dist sql rfc daily promotion diagram

### PR DESCRIPTION
**Pull request by andreimatei**
This change is [Reviewable](https://reviewable.io/reviews/raduberinde/cockroach/9)

### CODE REVIEW

File: docs/RFCS/distributed_sql.md [hunk 1]
```diff
@@ -607,6 +610,79 @@ TODO some examples, especially one where the intra-stream and inter-stream
...
+# A more complex example: Daily Promotion
+
+Let's draw a possible logical and physical plan for a more complex query. The
+point of the query is to help with a promotion that goes out daily, targetting
+customers that have spent over $1000 in the last year. We'll insert into the
+`DailyPromotion` table rows representing each such customer and the sum of her
+recent orders.
+
+```SQL
+INSERT INTO DailyPromotion FROM
```
Comment (RaduBerinde) [inline docs/RFCS/distributed_sql.md]: Should include table schemas and index info.
Also node that the query as written does not parse (missing `AS c`, missing `AS` for the second "table", `OrderSum` is not really needed, we can just do `HAVING SUM(OrderValue) > 1000` etc. The one in https://github.com/RaduBerinde/cockroach/wiki/Temp2 works (with those tables), you can just copy that stuff from there.

**Comment by RaduBerinde** [comment]
Sweet, the diagram is nice. Would be nice if we could make them all consistent or similar, but it's not that bad if they're not.

I think the inserers would need a "local" stage. They would need to generate the keys, or at least enough info for the routers to know what to do. They definitely are needed if there are indexes.

I think the benefit to moving the data on the target node before inserting is not that great compared to just doing the insertions from the node where we have the data. And there are difficulties, esp when indexes are involved. We will probably do it the simple way at least at first, so we could draw it like that too.

**Comment by andreimatei** [comment]
Fixed the diagram - inserter is now local stage only, we have a new IntentCollector aggregator

File: docs/RFCS/distributed_sql.md [hunk 2]
```diff
@@ -213,15 +213,16 @@ The grouping characteristic will be useful when we later decide how to
...
-`UNIQUE` for example). An aggregator with no grouping is a special but important
-case in which we are not aggregating multiple pieces of data, but we may be
-filtering, transforming, or reordering individual pieces of data.
+single-group aggregators (group key is the empty set of columns - `Group key:
+[]`, meaning everything is in the same group) which cannot be distributed. At
+the other end there are no-grouping aggregators which can be parallelized
+arbitrarily. Note that no-grouping aggregators are different than aggregators
+where the group key is the full set of columns - the latter still requires rows
+that are equal to be processed on a single node (this would be useful for an
+aggregator implementing `UNIQUE` for example). An aggregator with no grouping
```
Comment (RaduBerinde) [inline docs/RFCS/distributed_sql.md]: s/UNIQUE/DISTINCT/

File: docs/RFCS/distributed_sql.md [hunk 3]
```diff
@@ -859,6 +965,8 @@ KV integration involves three aspects:
 The details of all these need to be further investigated. Only 1 and 2 are
 required for M1; 3 is required for M2.
 
+TODO: Do we also need streaming reads from KV for `TableReader`?
```
Comment (RaduBerinde) [inline docs/RFCS/distributed_sql.md]: I think we can make do with the same method we use today. It's a potential improvement but not a requirement.

**Comment by RaduBerinde** [comment]
LGTM

File: docs/RFCS/distributed_sql.md (hunk 3 above)
Comment (andreimatei) [inline docs/RFCS/distributed_sql.md]: rephrased

**Pull request merged**